Haskell/Cabal.sublime-syntax

```diff
@@ -0,0 +1,406 @@
+    - include: value-end
+    - include: sequence-separators
+    - include: literal-urls
+
+###[ VERSIONS ]################################################################
+
+  version-keys:
+    - match: ^(\s*)({{version_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: version-values
+
+  version-values:
+    - meta_scope: meta.versions.cabal
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: operators
+    - include: literal-versions
+
+###[ OTHER TAGS ]##############################################################
+
+  other-keys:
+    - match: ^(\s*)({{identifier}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
+        3: meta.mapping.cabal punctuation.separator.mapping.key-value.cabal
+      push: other-values
+
+  other-values:
+    - meta_content_scope: meta.mapping.value.cabal
+    - include: value-end
+    - include: literal-emails
+
+  value-end:
+    - match: ^(?!\s*(?:--|$)|\1\s)
+      pop: 1
+
+###[ FUNCTION CALLS ]##########################################################
+
+  function-calls:
+    - match: '{{identifier}}(?=\()'
+      scope: meta.function-call.identifier.cabal variable.function.cabal
+      set:
+        - meta_include_prototype: false
+        - match: \(
+          scope: punctuation.section.group.begin.cabal
+          set: function-call-arguments
+
+  function-call-arguments:
+    - meta_scope: meta.function-call.arguments.cabal meta.group.cabal
+    - match: \)
+      scope: punctuation.section.group.end.cabal
+      pop: 1
+    - include: literal-versions
+    - include: operators
+    - include: ident-compilers
+    - include: ident-other
+
+###[ IDENTIFIERS ]#############################################################
+
+  ident-compilers:
+    - match: \b{{compiler_values}}\b
+      scope: constant.language.compiler.cabal
+
+  ident-namespaces:
+    - match: ({{module_identifier}})(\.)
+      captures:
+        1: variable.namespace.haskell
+        2: punctuation.accessor.dot.haskell
+
+  ident-modules:
+    - match: ({{module_identifier}})(\.)?
+      captures:
+        1: variable.namespace.haskell
+        2: punctuation.accessor.dot.haskell
+
+  ident-other:
+    - match: '{{identifier}}'
+
+###[ KEYWORDS AND OPERATORS ]##################################################
+
+  keywords:
+    - match: (if|else)\b
+      scope: keyword.control.conditional.cabal
+
+  operators:
+    - match: \|\||&&||!
+      scope: keyword.operator.logical.cabal
+    - match: \^?(>=|<=|==|[<>])
+      scope: keyword.operator.comparison.cabal
+
+  sequence-separators:
+    - match: ','
+      scope: punctuation.separator.sequence.cabal
+
+###[ LITERALS ]################################################################
+
+  literal-emails:
+    - match: <?(?=\w\S*?@\S+?\.\S+?)
```

Comment:
Is `+?` compliant with sregex?

Would branching make sense here?

Reply:
`+?` is compliant, yes. The contexts are borrowed from `Git Formats Common.sublime-syntax` and slightly modified to also highlight e-mail addresses without `<` `>`.

Yes, branching wouldn't work for plain e-mail addresses without angles.